### ISSUE TITLE
incorrect 'insufficient contrast' error for svg title attribute

### ISSUE THREAD

**Issue by lanfear** [pa11y/pa11y#416]
## Expected behaviour
Not an error because 'title' attribute of SVG is not a displayed element.

## Actual behaviour
Error: This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 1.34:1. Recommendation: change background to <#HEXRGB>.
WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail

<title>User Settings</title> (select with "html > body > div > tsc-site-header > div > tsc-site-header-sizing > div > div > div > tsc-settings-widget > div > button > svg > title")

## Steps to reproduce
Above error details may provide more insight into our potentially complex DOM layout if a simple test does not confirm, but repro should be to have an svg inside an element where text at default color would not pass contrast given the background of either the svg or parent element?

## Environment:
```
Pa11y:      5.0.4
Node.js:    8.9.0
npm:        5.5.1
OS:         10.0.15063 (win32)
```

## Notes:  As per MDN (https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title): 
```
When the current SVG document fragment is rendered as SVG on visual media, <title> element is not rendered as part of the graphics. However, some user agents may, for example, display the <title> element as a tooltip.
```
And, indeed, some browsers do present it as a tool-tip, but you cannot control the display of native browser tool-tip implementation so the error still seems irrelevant.

Here is chrome's reported entire 'computed' attribute list for the title element.

```
align-content:normal;
align-items:normal;
align-self:auto;
alignment-baseline:auto;
all:;
animation-delay:0s;
animation-direction:normal;
animation-duration:0s;
animation-fill-mode:none;
animation-iteration-count:1;
animation-name:none;
animation-play-state:running;
animation-timing-function:ease;
backface-visibility:visible;
background-attachment:scroll;
background-blend-mode:normal;
background-clip:border-box;
background-color:rgba(0, 0, 0, 0);
background-image:none;
background-origin:padding-box;
background-position-x:0%;
background-position-y:0%;
background-repeat-x:;
background-repeat-y:;
background-size:auto;
baseline-shift:0px;
block-size:auto;
border-bottom-color:rgb(0, 0, 0);
border-bottom-left-radius:0px;
border-bottom-right-radius:0px;
border-bottom-style:none;
border-bottom-width:0px;
border-collapse:separate;
border-image-outset:0px;
border-image-repeat:stretch;
border-image-slice:100%;
border-image-source:none;
border-image-width:1;
border-left-color:rgb(0, 0, 0);
border-left-style:none;
border-left-width:0px;
border-right-color:rgb(0, 0, 0);
border-right-style:none;
border-right-width:0px;
border-top-color:rgb(0, 0, 0);
border-top-left-radius:0px;
border-top-right-radius:0px;
border-top-style:none;
border-top-width:0px;
bottom:auto;
box-shadow:none;
box-sizing:border-box;
break-after:auto;
break-before:auto;
break-inside:auto;
buffered-rendering:auto;
caption-side:top;
caret-color:rgb(0, 0, 0);
clear:none;
clip:auto;
clip-path:none;
clip-rule:nonzero;
color:rgb(0, 0, 0);
color-interpolation:sRGB;
color-interpolation-filters:linearRGB;
color-rendering:auto;
column-count:auto;
column-fill:balance;
column-gap:normal;
column-rule-color:rgb(0, 0, 0);
column-rule-style:none;
column-rule-width:0px;
column-span:none;
column-width:auto;
contain:none;
content:normal;
counter-increment:none;
counter-reset:none;
cursor:pointer;
cx:0px;
cy:0px;
d:none;
direction:ltr;
display:inline;
dominant-baseline:auto;
empty-cells:show;
fill:rgb(255, 255, 255);
fill-opacity:1;
fill-rule:nonzero;
filter:none;
flex-basis:auto;
flex-direction:row;
flex-grow:0;
flex-shrink:1;
flex-wrap:nowrap;
float:none;
flood-color:rgb(0, 0, 0);
flood-opacity:1;
font-family:proxima-nova, Helvetica, Arial, sans-serif;
font-feature-settings:normal;
font-kerning:auto;
font-size:16px;
font-stretch:100%;
font-style:normal;
font-variant-caps:normal;
font-variant-east-asian:normal;
font-variant-ligatures:normal;
font-variant-numeric:normal;
font-variation-settings:normal;
font-weight:400;
grid-auto-columns:auto;
grid-auto-flow:row;
grid-auto-rows:auto;
grid-column-end:auto;
grid-column-start:auto;
grid-row-end:auto;
grid-row-start:auto;
grid-template-areas:none;
grid-template-columns:none;
grid-template-rows:none;
height:auto;
hyphens:manual;
image-rendering:auto;
inline-size:auto;
isolation:auto;
justify-content:normal;
justify-items:normal;
justify-self:auto;
left:auto;
letter-spacing:normal;
lighting-color:rgb(255, 255, 255);
line-break:auto;
line-height:normal;
list-style-image:none;
list-style-position:outside;
list-style-type:disc;
margin-bottom:0px;
margin-left:0px;
margin-right:0px;
margin-top:0px;
marker-end:none;
marker-mid:none;
marker-start:none;
mask:none;
mask-type:luminance;
max-block-size:none;
max-height:none;
max-inline-size:none;
max-width:none;
min-block-size:0px;
min-height:0px;
min-inline-size:0px;
min-width:0px;
mix-blend-mode:normal;
object-fit:fill;
object-position:50% 50%;
offset-distance:0px;
offset-path:none;
offset-rotate:auto 0deg;
opacity:1;
order:0;
orphans:2;
outline-color:rgb(0, 0, 0);
outline-offset:0px;
outline-style:none;
outline-width:0px;
overflow-anchor:auto;
overflow-wrap:normal;
overflow-x:visible;
overflow-y:visible;
overscroll-behavior-x:auto;
overscroll-behavior-y:auto;
padding-bottom:0px;
padding-left:0px;
padding-right:0px;
padding-top:0px;
page:;
paint-order:fill stroke markers;
perspective:none;
perspective-origin:50% 50%;
pointer-events:auto;
position:static;
quotes:;
r:0px;
resize:none;
right:auto;
row-gap:normal;
rx:auto;
ry:auto;
scroll-behavior:auto;
shape-image-threshold:0;
shape-margin:0px;
shape-outside:none;
shape-rendering:auto;
size:;
speak:normal;
stop-color:rgb(0, 0, 0);
stop-opacity:1;
stroke:none;
stroke-dasharray:none;
stroke-dashoffset:0px;
stroke-linecap:butt;
stroke-linejoin:miter;
stroke-miterlimit:4;
stroke-opacity:1;
stroke-width:1px;
tab-size:8;
table-layout:auto;
text-align:center;
text-align-last:auto;
text-anchor:start;
text-combine-upright:none;
text-decoration-color:rgb(0, 0, 0);
text-decoration-line:none;
text-decoration-skip-ink:auto;
text-decoration-style:solid;
text-indent:0px;
text-orientation:mixed;
text-overflow:clip;
text-rendering:auto;
text-shadow:none;
text-size-adjust:100%;
text-transform:none;
text-underline-position:auto;
top:auto;
touch-action:auto;
transform:none;
transform-box:view-box;
transform-origin:0px 0px;
transform-style:flat;
transition-delay:0s;
transition-duration:0s;
transition-property:all;
transition-timing-function:ease;
unicode-bidi:normal;
user-select:auto;
vector-effect:none;
vertical-align:baseline;
visibility:visible;
white-space:normal;
widows:2;
width:auto;
will-change:auto;
word-break:normal;
word-spacing:0px;
word-wrap:normal;
writing-mode:horizontal-tb;
x:0px;
y:0px;
z-index:auto;
zoom:1;
-webkit-app-region:none;
-webkit-appearance:none;
-webkit-border-after-color:rgb(0, 0, 0);
-webkit-border-after-style:none;
-webkit-border-after-width:0px;
-webkit-border-before-color:rgb(0, 0, 0);
-webkit-border-before-style:none;
-webkit-border-before-width:0px;
-webkit-border-end-color:rgb(0, 0, 0);
-webkit-border-end-style:none;
-webkit-border-end-width:0px;
-webkit-border-horizontal-spacing:0px;
-webkit-border-start-color:rgb(0, 0, 0);
-webkit-border-start-style:none;
-webkit-border-start-width:0px;
-webkit-border-vertical-spacing:0px;
-webkit-box-align:stretch;
-webkit-box-decoration-break:slice;
-webkit-box-direction:normal;
-webkit-box-flex:0;
-webkit-box-ordinal-group:1;
-webkit-box-orient:horizontal;
-webkit-box-pack:start;
-webkit-box-reflect:none;
-webkit-font-size-delta:;
-webkit-font-smoothing:antialiased;
-webkit-highlight:none;
-webkit-hyphenate-character:auto;
-webkit-line-clamp:none;
-webkit-locale:"en-US";
-webkit-logical-height:auto;
-webkit-logical-width:auto;
-webkit-margin-after:0px;
-webkit-margin-after-collapse:collapse;
-webkit-margin-before:0px;
-webkit-margin-before-collapse:collapse;
-webkit-margin-bottom-collapse:collapse;
-webkit-margin-end:0px;
-webkit-margin-start:0px;
-webkit-margin-top-collapse:collapse;
-webkit-mask-box-image-outset:0px;
-webkit-mask-box-image-repeat:stretch;
-webkit-mask-box-image-slice:0 fill;
-webkit-mask-box-image-source:none;
-webkit-mask-box-image-width:auto;
-webkit-mask-clip:border-box;
-webkit-mask-composite:source-over;
-webkit-mask-image:none;
-webkit-mask-origin:border-box;
-webkit-mask-position-x:0%;
-webkit-mask-position-y:0%;
-webkit-mask-repeat-x:;
-webkit-mask-repeat-y:;
-webkit-mask-size:auto;
-webkit-max-logical-height:none;
-webkit-max-logical-width:none;
-webkit-min-logical-height:0px;
-webkit-min-logical-width:0px;
-webkit-padding-after:0px;
-webkit-padding-before:0px;
-webkit-padding-end:0px;
-webkit-padding-start:0px;
-webkit-perspective-origin-x:;
-webkit-perspective-origin-y:;
-webkit-print-color-adjust:economy;
-webkit-rtl-ordering:logical;
-webkit-ruby-position:before;
-webkit-tap-highlight-color:rgba(0, 0, 0, 0.18);
-webkit-text-combine:none;
-webkit-text-decorations-in-effect:none;
-webkit-text-emphasis-color:rgb(0, 0, 0);
-webkit-text-emphasis-position:over right;
-webkit-text-emphasis-style:none;
-webkit-text-fill-color:rgb(0, 0, 0);
-webkit-text-security:none;
-webkit-text-stroke-color:rgb(0, 0, 0);
-webkit-text-stroke-width:0px;
-webkit-transform-origin-x:;
-webkit-transform-origin-y:;
-webkit-transform-origin-z:;
-webkit-user-drag:auto;
-webkit-user-modify:read-only;
-webkit-border-image:none;
-webkit-text-orientation:vertical-right;
```

**Comment by hollsk** [comment]
Hi @lanfear - thanks for using Pa11y! 

I opened an issue in the HTML_Codesniffer repo for this - please follow that one for updates: https://github.com/squizlabs/HTML_CodeSniffer/issues/229

**Comment by thibaudcolas** [comment]
This now seems to be released in HTML_CodeSniffer 2.30 :) See also #419.

**Comment by joeyciechanowicz** [comment]
HTML_CodeSniffer is now imported from NPM we're using v2.3.0. Release of pa11y with this version will be coming soon.
